src/core/manager.c

```diff
@@ -3455,12 +3456,11 @@ void manager_override_watchdog(Manager *m, WatchdogType t, usec_t timeout) {
         if (m->watchdog_overridden[t] == timeout)
                 return;
 
-        if (t == WATCHDOG_RUNTIME) {
-                usec_t usec = timestamp_is_set(timeout) ? timeout : m->watchdog[t];
-
+        usec = (timeout == USEC_INFINITY) ? m->watchdog[t] : timeout;
+        if (t == WATCHDOG_RUNTIME)
                 (void) watchdog_setup(usec);
-        } else if (t == WATCHDOG_PRETIMEOUT)
-                (void) watchdog_setup_pretimeout(timeout);
+        else if (t == WATCHDOG_PRETIMEOUT)
+                (void) watchdog_setup_pretimeout(usec);
```

Comment:
this changes behaviour too - before, `timeout` was always used, now `m->watchdog[t]` will be used instead

Reply:
This one is intentional. I think should be able to clear the override to use the original value. So the behavior is the same as the regular WATCHDOG_RUNTIME timeout.